internal/services/appservice/service_plan_resource.go

```diff
@@ -248,7 +248,11 @@ func (r ServicePlanResource) Read() sdk.ResourceFunc {
 				// sku read
 				if sku := model.Sku; sku != nil {
 					if sku.Name != nil {
-						state.Sku = *sku.Name
+						for _, item := range helpers.AllKnownServicePlanSkus() {
+							if strings.EqualFold(*sku.Name, item) {
```

Comment:
We tend to want to use DiffSuppressFunc rather than custom code in the read function. Could we do that here instead or is there a reason we're going this route?